k9mail/src/main/java/com/fsck/k9/activity/compose/RecipientPresenter.java

```diff
@@ -766,6 +767,51 @@ public void messageReceived() {
         }
     }
 
+    public void handleResetPartnerKeyResult() {
+        loadPEpStatus();
+    }
+
+    public boolean canResetSenderKeys(
+            List<Address> newToAdresses,
+            List<Address> newCcAdresses,
+            List<Address> newBccAdresses
+    ) {
+        return recipientConditionsForKeyReset(newToAdresses, newCcAdresses, newBccAdresses)
+                && ratingConditionsForSenderKeyReset();
+    }
+
+    private void handleSingleAddressKeyResetAllowance(
+            List<Address> newToAdresses,
+            List<Address> newCcAdresses,
+            List<Address> newBccAdresses
+    ) {
+        if (canResetSenderKeys(newToAdresses, newCcAdresses, newBccAdresses)) {
+            recipientMvpView.showResetPartnerKeyOption();
+        } else {
+            recipientMvpView.hideResetPartnerKeyOption();
+        }
+    }
+
+    private boolean recipientConditionsForKeyReset(
+            List<Address> newToAdresses,
+            List<Address> newCcAdresses,
+            List<Address> newBccAdresses
+    ) {
+        return account != null && account.isPlanckPrivacyProtected()
+                && newToAdresses.size() == ONE_ADDRESS
+                && newBccAdresses.isEmpty()
+                && newCcAdresses.isEmpty()
+                && !Preferences.getPreferences(context).containsAccountByEmail(newToAdresses.get(0).getAddress());
```

Comment:
what is this last condition doing? We check if there is only 1 address in to: in the compose dialog to show the reset option and then we check here if that address is not in a specific preference?

Reply:
`Preferences` is a singleton used to load and access accounts.
So I just check that the email address does not belong to one of the user's account so they will not key-reset their own accounts.

Reply:
perfect clear